src/fr/scanmanga/src/eu/kanade/tachiyomi/extension/fr/scanmanga/ScanManga.kt

```diff
@@ -78,119 +86,87 @@ class ScanManga : ParsedHttpSource() {
         return SManga.create().apply {
             title = element.select("a.nom_manga").text()
             setUrlWithoutDomain(element.select("a.nom_manga").attr("href"))
-            /*thumbnail_url = element.select(".logo_manga img").let {
-                if (it.hasAttr("data-original"))
-                    it.attr("data-original") else it.attr("src")
-            }*/
-            // Better not use it, width is too large, which results in terrible image
         }
     }
 
     override fun latestUpdatesNextPageSelector(): String? = null
 
     // Search
-    override fun searchMangaFromElement(element: Element) = throw UnsupportedOperationException()
-
-    override fun searchMangaNextPageSelector() = throw UnsupportedOperationException()
-
-    override fun searchMangaParse(response: Response): MangasPage = parseMangaFromJson(response)
-
-    private fun shuffle(s: String?): String {
-        val result = StringBuffer(s!!)
-        var n = result.length
-        while (n > 1) {
-            val randomPoint: Int = Random.nextInt(n)
-            val randomChar = result[randomPoint]
-            result.setCharAt(n - 1, randomChar)
-            n--
-        }
-        return result.toString()
-    }
-
     override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
-        val searchHeaders = headersBuilder()
-            .add("Referer", "$baseUrl/scanlation/liste_series.html")
-            .add("x-requested-with", "XMLHttpRequest")
+        val url = desktopBaseUrl.toHttpUrlOrNull()!!.newBuilder()
+            .addPathSegments("scanlation/liste_series.html")
+            .addQueryParameter("q", query)
             .build()
-
-        return GET("$baseUrl/scanlation/scan.data.json", searchHeaders)
+            .toString()
+
+        return GET(
+            url,
+            headersBuilder()
+                .add("User-Agent", "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/115.0 Safari/537.36")
+                .add("Accept-Language", "fr-FR,fr;q=0.9,en-US;q=0.8,en;q=0.7")
+                .add("Referer", desktopBaseUrl)
```

Comment:
Browsers always add a trailing slash:
```suggestion
                .add("Referer", "$desktopBaseUrl/")
```